src/QueryPipeline/QueryPipelineBuilder.cpp

```diff
@@ -344,15 +347,45 @@ std::unique_ptr<QueryPipelineBuilder> QueryPipelineBuilder::joinPipelines(
     ///                   ╞> FillingJoin ─> Resize ╣     ╞> Joining ─> (totals)
     /// (totals) ─────────┘                        ╙─────┘
 
-    size_t num_streams = left->getNumStreams();
-    right->resize(1);
+    auto num_streams = left->getNumStreams();
 
-    auto adding_joined = std::make_shared<FillingRightJoinSideTransform>(right->getHeader(), join);
-    InputPort * totals_port = nullptr;
-    if (right->hasTotals())
-        totals_port = adding_joined->addTotalsPort();
+    if (join->supportParallelJoin())
+    {
+        if (!keep_left_read_in_order)
+        {
+            left->resize(max_streams);
+            num_streams = max_streams;
+        }
+
+        if (!right->hasTotals())
```

Comment:
Why not `join->supportParallelJoin() && !right->hasTotals()` ?

Reply:
OK